xesmf/frontend.py

```diff
@@ -314,9 +323,19 @@ def __init__(
         self.weights = read_weights(weights, self.n_in, self.n_out)
 
         # replace zeros by NaN in mask
-        if self.grid_out.mask is not None and self.grid_out.mask[0] is not None:
+        if default_weights is True:
```

Comment:
this does not reproduce the original behavior! if the mask is defined, `add_nans_to_weights` should be applied.
I would prefer something along the lines of 

```python
if ((self.grid_out.mask is not None) and (self.grid_out.mask[0] is not None)) or default_weights is True:
```

Reply:
That however does not allow full control over whether to apply add_nans_to_weights or not.
But as a `default_weights=True` setting cannot be overruled it is still fine for my purpose.